x-pack/solutions/security/plugins/elastic_assistant/server/routes/defend_insights/get_defend_insight.ts

```diff
@@ -86,14 +89,27 @@ export const getDefendInsightRoute = (router: IRouter<ElasticAssistantRequestHan
             });
           }
 
-          const defendInsight = await updateDefendInsightLastViewedAt({
+          const defendInsights = await dataClient.findDefendInsightsByParams({
```

Comment:
Extracted `dataClient.findDefendInsightsByParams` from `updateDefendInsightLastViewedAt`. This should be sufficient for our use case. If any `endpointIds` are from outside the current space, we throw early. Otherwise, we proceed and reuse the already fetched insights in the following steps.